pkg/volume/scaleio/sio_mgr.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaleio
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/golang/glog"
+
+	siotypes "github.com/codedellemc/goscaleio/types/v1"
+)
+
+type storageInterface interface {
+	CreateVolume(string, int64) (*siotypes.Volume, error)
+	AttachVolume(string) (string, error)
+	IsAttached(string) (bool, error)
+	DetachVolume(string) error
+	DeleteVolume(string) error
+}
+
+type sioMgr struct {
+	client     sioInterface
+	configData map[string]string
+}
+
+func newSioMgr(configs map[string]string) (*sioMgr, error) {
+	if configs == nil {
+		return nil, errors.New("missing configuration data")
+	}
+	configs[confKey.protectionDomain] = defaultString(configs[confKey.protectionDomain], "default")
```

Comment:
@rootfs Yes, I used `defaults.go` for the ScaleIO VolumeSource.  Will investigate its usage for `StorageClass` as well.